### PR TITLE
Bugsnag

### PR DESCRIPTION
**Pull request by mcasper**
A rough outline for the Bugsnag component, does not enable it in the app store. We can merge this and then iterate on the design and refactors.

Is based off of the basecamp_redux branch, so @mikeastock should probably merge his branch before this one.

screen shot 2015-09-13 at 10 08 01 pm

### CODE REVIEW

File: app/services/bugsnag/client.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+class Bugsnag::Client
```
Comment (mcasper) [inline app/services/bugsnag/client.rb]: This isn't used yet, which is why some of the methods are mocked, and there are no tests

File: frontend/root.js [hunk 2]
```diff
@@ -23,6 +24,9 @@ export default class Root extends Component {
         <Provider store={store}>
           { () => <BasecampApp /> }
         </Provider>
+        <Provider store={store}>
+          { () => <BugsnagApp /> }
```
Comment (mikeastock) [inline frontend/root.js]: I believe both components can be in the same Provider

Comment (mikeastock) [inline frontend/root.js]:
```js 
<div>
        <Provider store={store}>
          { () => {
            <BasecampApp /> 
            <BugsnagApp /> 
            }
        </Provider>
        {this._renderDevTools()}
      </div>
```

Comment (mcasper) [inline frontend/root.js]: The console got very mad at me when I tried that

Comment (mcasper) [inline frontend/root.js]: Oh, hmm. I didn't try it exactly like that.

Comment (mcasper) [inline frontend/root.js]: No, it still doesn't want that

**Pull request merged**